src/NServiceBus.TransactionalSession/TransactionalSession.cs

```diff
@@ -76,6 +82,14 @@ protected override void Setup(FeatureConfigurationContext context)
                 "Transaction commit control message delay acknowledgement behavior");
         }
 
+        /// <summary>
+        /// Sets up the necessary infrastructure.
+        /// </summary>
+        /// <param name="context">The feature configuration context.</param>
+        protected virtual void SetupCore(FeatureConfigurationContext context)
```

Comment:
what's the benefit over just overriding `Setup` on a specific implementation too?

Reply:
I don't really mind that much. I was just thinking with that the order is guaranteed and controlled by the base feature and then as an inheritor you don't have to worry about that. But you could argue having the ability to call the base method where ever might also have benefits. Will change